front/src/modules/timetableItem/components/ImportTimetableItem/helpers/parseXML.ts

```diff
@@ -72,6 +91,175 @@ export const getMostFrequentInterval = (schedules: TrainSchedule[]): Duration =>
+    ...trainScheduleToPacedTrain(
+      modelTrainSchedule,
+      pacedTrainId,
+      intervalDuration,
+      totalTimeWindow
+    ),
+    exceptions: [],
+  };
+
+  const osrdDefaultOccurrences: {
+    startTime: Date;
+    matchedImportedSchedule: TrainSchedule | null;
+  }[] = [];
+  for (let i = 0; i < numberOfExpectedOccurrences; i += 1) {
+    const expectedDepartureTime = addDurationToDate(
+      firstImportedDepartureTime,
+      new Duration({ minutes: i * intervalMinutes })
+    );
+    osrdDefaultOccurrences.push({
+      startTime: expectedDepartureTime,
+      matchedImportedSchedule: null,
+    });
+  }
+
+  const matchedImportedScheduleKeys = new Set<string>();
+  const getScheduleKey = (s: TrainSchedule) => `${s.start_time}-${s.train_name}`;
+
+  // Match OSRD Default Occurrences to imported Trains
+  osrdDefaultOccurrences.forEach((osrdOccurrence) => {
+    let bestCandidate: TrainSchedule | null = null;
+    let minTimeDifference = Infinity;
+
+    const availableImportedSchedules = sortedImportedSchedules.filter(
+      (schedule) => !matchedImportedScheduleKeys.has(getScheduleKey(schedule))
+    );
+
+    availableImportedSchedules.forEach((importedSchedule) => {
+      const importedDepartureTime = new Date(importedSchedule.start_time);
+      const timeDifferenceMs = Math.abs(
+        osrdOccurrence.startTime.getTime() - importedDepartureTime.getTime()
+      );
+      const timeDifferenceMinutes = timeDifferenceMs / (1000 * 60);
+
+      if (
+        timeDifferenceMinutes <= intervalMinutes / 4 &&
+        timeDifferenceMinutes < minTimeDifference
+      ) {
+        bestCandidate = importedSchedule;
+        minTimeDifference = timeDifferenceMinutes;
+      }
+    });
+
+    if (bestCandidate) {
+      osrdOccurrence.matchedImportedSchedule = bestCandidate;
+      matchedImportedScheduleKeys.add(getScheduleKey(bestCandidate));
+    }
+  });
+
+  const exceptions: PacedTrainException[] = [];
+  osrdDefaultOccurrences.forEach((osrdOccurrence, index) => {
+    if (!osrdOccurrence.matchedImportedSchedule) {
+      // Case 1: OSRD default occurrence has no correspondence in the imported file
+      // Create a disabled exception
+      exceptions.push({
+        key: uuidV4(),
+        occurrence_index: index,
+        disabled: true,
+      });
+    } else {
+      // Case 2: OSRD default occurrence has a correspondence in imported file
+      const updatedOccurrence = trainScheduleToPacedTrain(
+        osrdOccurrence.matchedImportedSchedule,
+        pacedTrainId,
+        intervalDuration,
+        totalTimeWindow
+      );
+
+      const baseException = generatePacedTrainException(
+        updatedOccurrence,
+        originalPacedTrain,
+        index
+      );
+
+      const exception: PacedTrainException = {
+        ...baseException,
+        key: uuidV4(),
+        occurrence_index: index,
+      };
+
+      const cleanException = omit(exception, 'train_name');
+      exceptions.push(cleanException);
```

Comment:
Shouldn't we check whether the exception is empty? If `baseException` contains only `train_name`, we can skip it?